src/platform/Linux/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -255,19 +453,34 @@ void LinuxWiFiDriver::ScanNetworks(ByteSpan ssid, WiFiDriver::ScanCallback * cal
 
 size_t LinuxWiFiDriver::WiFiNetworkIterator::Count()
 {
-    return driver->mStagingNetwork.Empty() ? 0 : 1;
+    size_t count = 0;
+
+    for (uint8_t networkIndex = 0; networkIndex < kMaxNetworks; networkIndex++)
+    {
+        if (!driver->mStagingNetworks[networkIndex].Empty())
+        {
+            count++;
```

Comment:
Why are we bothering with `mCurrentNetId` then?  How does that differ from the count of non-empty networks?

Reply:
Yeah the function can return mCurrentNetid instead of looping over mStagingNetworks to find non empty networks.